core/src/main/java/org/apache/carbondata/core/scan/filter/executer/RangeValueFilterExecuterImpl.java

```diff
@@ -242,7 +256,14 @@ public boolean isScanRequired(byte[] blockMinValue, byte[] blockMaxValue, byte[]
     //                       Block Min <-----------------------> Block Max
     //         Filter Min <-----------------------------------------------> Filter Max
 
-    if (isDimensionPresentInCurrentBlock == true) {
+    if (isDimensionPresentInCurrentBlock) {
+      // if a column is unsorted, min and max for that column will not be correct and hence
```

Comment:
During data loading even if data is not sorted we are calculating the min max so not need to add this code, in all the filter classes only if inverted index is false we need to go iterative approach